src/encode.rs

```diff
@@ -56,11 +56,14 @@ pub fn encode<T: AsRef<[u8]>>(input: T) -> String {
 pub fn encode_engine<E: Engine, T: AsRef<[u8]>>(input: T, engine: &E) -> String {
     let encoded_size = encoded_len(input.as_ref().len(), engine.config().encode_padding())
         .expect("integer overflow when calculating buffer size");
-    let mut buf = vec![0; encoded_size];
+    let mut buf = Vec::with_capacity(encoded_size);
+    unsafe {
+        buf.set_len(encoded_size);
```

Comment:
The safety documentation of [`set_len`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.set_len) is pretty clear that this is UB. Namely it states
> The elements at `old_len..new_len` must be initialized.

I think it might be safe to call *after* `encode_with_padding`.